common/src/main/scala/org/apache/celeborn/common/util/ThreadUtils.scala

```diff
@@ -299,3 +303,10 @@ object ThreadUtils {
     }
   }
 }
+
+class ThreadExceptionHandler(executorService: String)
+  extends Thread.UncaughtExceptionHandler with Logging {
+
+  override def uncaughtException(t: Thread, e: Throwable): Unit =
+    log.error("Uncaught exception in executor service {}, thread {}", executorService, t, e)
```

Comment:
always use thunk in logging
```suggestion
    logError("Uncaught exception in executor service {}, thread {}", executorService, t, e)
```